### ISSUE TITLE
How to cache go deps in multi-stage builds?

### ISSUE THREAD

**Issue by rolandjitsu** [docker/buildx#357]
Probably not the place to ask this, but I'm a little confused about `cache-to`/`cache-from` logic.

I was looking at [fsutil/Dockerfile](https://github.com/tonistiigi/fsutil/blob/master/Dockerfile) and I see the use of:
```dockerfile
RUN --mount=target=. \
    --mount=target=/root/.cache,type=cache \
```
How does this tie into the `cache-to`/`cache-from`? Am I supposed to use the `cache-to`/`cache-from`? Or are the mount statements independent of that?

I cannot seem to find some good docs on how to properly cache docker builds when using the `docker buildx bake -o /some-dir` to build binaries.

**Comment by rolandjitsu** [comment]
I think I found more docs on this. Found [hack/dockerfiles/gomod.Dockerfile](https://github.com/tonistiigi/fsutil/blob/master/hack/dockerfiles/gomod.Dockerfile).

**Comment by thaJeztah** [comment]
Yes, (cache) mounts are explicitly excluded from the layer, and as such also not part of the build-cache. They help with repeated builds on the same machine, but you can't share them with other machines.
